pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go

```diff
@@ -158,44 +159,56 @@ func syncVolume(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCl
 	currentPhase := volume.Status.Phase
 	nextPhase := currentPhase
 
+	_, exists, err := volumeIndex.Get(volume)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		volumeIndex.Add(volume)
+	}
+
 	switch currentPhase {
-	// pending volumes are available only after indexing in order to be matched to claims.
 	case api.VolumePending:
 		if volume.Spec.ClaimRef != nil {
-			// Pending volumes that have a ClaimRef were recently recycled.  The Recycler set the phase to VolumePending
-			// to start the volume again at the beginning of this lifecycle.
-			// ClaimRef is the last bind between persistent volume and claim.
-			// The claim has already been deleted by the user at this point
-			oldClaimRef := volume.Spec.ClaimRef
-			volume.Spec.ClaimRef = nil
-			_, err = binderClient.UpdatePersistentVolume(volume)
+			claim, err := binderClient.GetPersistentVolumeClaim(volume.Spec.ClaimRef.Namespace, volume.Spec.ClaimRef.Name)
 			if err != nil {
```

Comment:
```go
if errors.IsNotFound(err){
  // stuff
} else if err != nil{
  // other stuff
}
```

That keeps me from having to scan down and follow the nesting of multiple `else`s to sort out what's happening.